h5io_browser/base.py

```diff
@@ -229,6 +230,44 @@ def _merge_nested_dict(main_dict: dict, add_dict: dict) -> dict:
             main_dict[k] = v
     return main_dict
 
+from collections import defaultdict
+_FILE_HANDLE_CACHE_COUNTER = defaultdict(int)
+_FILE_HANDLE_CACHE = {}
+import threading
+_FILE_HANDLE_CACHE_LOCK = threading.RLock()
```

Comment:
Maybe we should forget about threading for the time being. The h5 library by itself must be compiled in a special mode to be thread-save, and the conda-version usually isn't.

Reply:
Ah, didn't know about that.  Anyway, the chances of pyiron becoming thread safe in the near future are comfortably close to zero